Integrations/integration-ESM-v10.yml

```diff
@@ -302,17 +551,229 @@ script:
+            'Contents': fixed_organizations,
+            'ReadableContentsFormat': formats['markdown'],
+            'HumanReadable': tblToMd('Case Events:', fixed_organizations, headers),
+            'EntryContext': {}
+       }
+
+    def events_to_entry(events):
+        if not events:
+            return 'No events were found'
+
+        headers = ['ID', 'LastTime', 'Message']
+        fixed_events = []
+        for event in events:
+            fixed_events.append({
+                'ID' : event['id']['value'].replace('|', '\|'),
+                'LastTime' : event['lastTime'],
+                'Message' : event['message'],
+            })
+
+        context = {'CaseEvents(val.ID && val.ID == obj.ID)' : fixed_events}
+        return {
+            'ContentsFormat': formats['json'],
+            'Type': entryTypes['note'],
+            'Contents': fixed_events,
+            'ReadableContentsFormat': formats['markdown'],
+            'HumanReadable': tblToMd('Case Events:', fixed_events, headers),
+            'EntryContext': context
+       }
+
+    def users_to_entry(users):
+        # unreachable code - in order to send command, one must be logged in. therefore there is at least one user.
+        if not users:
+            return 'No users were found'
+
+        headers = ['ID', 'Name', 'Email', 'SMS', 'IsMaster', 'IsAdmin']
+        fixed_users = []
+        for user in users:
+            fixed_users.append({
+                'ID' : user['id']['value'],
+                'Name' : user['username'],
+                'Email' : user['email'],
+                'SMS' : user['sms'],
+                'IsMaster' : user['master'],
+                'IsAdmin' : user['admin'],
+            })
+
+        context = {'EsmUser(val.ID && val.ID == obj.ID)' : fixed_users}
+        return {
+            'ContentsFormat': formats['json'],
+            'Type': entryTypes['note'],
+            'Contents': fixed_users,
+            'ReadableContentsFormat': formats['markdown'],
+            'HumanReadable': tblToMd('Users:', fixed_users, headers),
+            'EntryContext': context
+       }
+
     esm = NitroESM(demisto.params()['ip']+':'+demisto.params()['port'], demisto.params()['credentials']['identifier'], demisto.params()['credentials']['password'])
+    final_result = 'No result set'
 
-    if demisto.command() == 'test-module':
-        esm.fetchAllFields()
-        demisto.results('ok')
+    if demisto.command() == 'fetch-incidents':
+        last_run = demisto.getLastRun()
+        # for backward compatibility uses
+        if last_run.has_key('value') and not last_run.has_key('alarms'):
+            last_run['alarms'] = last_run['value']
+
+        start_alarms = last_run.get('alarms', (datetime.datetime.now() - datetime.timedelta(minutes=1)).isoformat())
+        start_cases = last_run.get('cases', int(demisto.params()['startingCaseID']))
+
+        incidents = []
+        mode = demisto.params()['fetchTypes'].lower()
```

Comment:
If user upgrade existing instance from old ESM to this ESM - I think `fetchTypes` will not get the default value of `alarms`

I think we should have fallback to `alarms` also here